packages/SwingSet/src/liveslots/collectionManager.js

```diff
@@ -351,16 +370,11 @@ export function makeCollectionManager(
     }
 
     function init(key, value) {
-      matches(key, keyShape) ||
-        assert.fail(X`invalid key type for collection ${q(label)}`);
+      fit(key, keyShape, keyLabel);
       !has(key) ||
         assert.fail(X`key ${key} already registered in collection ${q(label)}`);
-      if (valueShape) {
-        matches(value, valueShape) ||
-          assert.fail(X`invalid value type for collection ${q(label)}`);
-      }
       currentGenerationNumber += 1;
```

Comment:
@FUDCo this is the change I'm most worried about. If the pattern does not match, the old code would throw before 
```js
currentGenerationNumber += 1;
```
whereas the new code fails after. We could swap the lines, but then the old code would increment before serialization and the new code would increment afterwards. I don't know the significance of `currentGenerationNumber` so I can't judge. Please advise.

Reply:
Nevermind. I'm switching the order. Serialization before increment is harmless.